ARROW-16170: [Java][Docs] Synch java code tutorial with java cookbook (#183)
apache/arrow-cookbook: java/source/schema.rst

Working with Schema
===================

> Let's start talk about tabular data. Data often comes in the form of two-dimensional
> sets of heterogeneous data (such as database tables, CSV files...). Arrow provides
> several abstractions to handle such data conveniently and efficiently.

Reviewer: ```suggestion
Let's start talking about tabular data. Data often comes in the form of two-dimensional
sets of heterogeneous data (such as database tables, CSV files...). Arrow provides
several abstractions to handle such data conveniently and efficiently.
```

.. contents::

> Creating Field
> ==============

Reviewer: ```suggestion
Creating Fields
===============
```

    points: List<intCol: Int(32, true)>

> Adding Metadata for Field
> =========================

Reviewer: ```suggestion
Adding Metadata to Fields
=========================
```

    {A=Id card, B=Passport, C=Visa}

Creating the Schema
===================

> A Schema describe a sequence of columns in tabular data.

Reviewer: ```suggestion
A schema describes a sequence of columns in tabular data, and consists
of a list of fields.
```

    import org.apache.arrow.vector.types.pojo.ArrowType;
    import org.apache.arrow.vector.types.pojo.Field;
    import org.apache.arrow.vector.types.pojo.FieldType;
    import java.util.ArrayList;
    import java.util.List;

    Field name = new Field("name", FieldType.nullable(new ArrowType.Utf8()), null);
    Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), null), null);
    Field age = new Field("age", FieldType.nullable(new ArrowType.Int(32, true)), null);
    FieldType intType = new FieldType(true, new ArrowType.Int(32, true), /*dictionary=*/null);
    FieldType listType = new FieldType(true, new ArrowType.List(), /*dictionary=*/null);
    Field childField = new Field("intCol", intType, null);
    List<Field> childFields = new ArrayList<>();
    childFields.add(childField);
    Field points = new Field("points", listType, childFields);
    Schema schemaPerson = new Schema(asList(name, document, age, points));

    System.out.print(schemaPerson);

.. testoutput::

    Schema<name: Utf8, document: Utf8, age: Int(32, true), points: List<intCol: Int(32, true)>>

> Adding Metadata for Schema

Reviewer: ```suggestion
Adding Metadata to Schemas
```

Reviewer: IMO, we can just have one section for both field and schema metadata instead of separating it.

Author: Added

==========================

In case we need to add metadata to our definition we could use:

.. testcode::

    import org.apache.arrow.vector.types.pojo.Schema;

>     import static java.util.Arrays.asList;

Reviewer: Can we sort the imports?

Author: Sorted

    import java.util.ArrayList;
    import java.util.HashMap;
    import java.util.List;
    import java.util.Map;

    Field name = new Field("name", FieldType.nullable(new ArrowType.Utf8()), null);
    Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), null), null);
    Field age = new Field("age", FieldType.nullable(new ArrowType.Int(32, true)), null);
    FieldType intType = new FieldType(true, new ArrowType.Int(32, true), /*dictionary=*/null);
    FieldType listType = new FieldType(true, new ArrowType.List(), /*dictionary=*/null);
    Field childField = new Field("intCol", intType, null);
    List<Field> childFields = new ArrayList<>();
    childFields.add(childField);
    Field points = new Field("points", listType, childFields);
    Map<String, String> metadataSchema = new HashMap<>();
    metadataSchema.put("Key-1", "Value-1");
    Schema schemaPerson = new Schema(asList(name, document, age, points), metadataSchema);

    System.out.print(schemaPerson);

.. testoutput::

    Schema<name: Utf8, document: Utf8, age: Int(32, true), points: List<intCol: Int(32, true)>>(metadata: {Key-1=Value-1})

Populate Data
=============

> Let's populate a `VectorSchemaRoot` with a small batch of records:

Reviewer: ```suggestion
Let's populate a ``VectorSchemaRoot`` with a small batch of records:
```
